File: regions/system/library/swift/tracing/Sources/K1s0Tracing/Propagation.swift
public func injectContext(into headers: inout [String: String], context: TraceContext, baggage: Baggage? = nil) {
    headers["traceparent"] = context.toTraceparent()
    if let baggage = baggage {
        let header = baggage.toHeader()
        if !header.isEmpty {
            headers["baggage"] = header
        }
    }
}

public func extractContext(from headers: [String: String]) -> (context: TraceContext?, baggage: Baggage) {
    let context = TraceContext.fromTraceparent(headers["traceparent"] ?? "")
    let baggage = Baggage.fromHeader(headers["baggage"] ?? "")
    return (context, baggage)
}

File: regions/system/library/csharp/tracing/src/K1s0.Tracing/Propagation.cs
namespace K1s0.Tracing;

public static class Propagation
{
    public static void InjectContext(IDictionary<string, string> headers, TraceContext ctx, Baggage? baggage = null)
    {
        headers["traceparent"] = ctx.ToTraceparent();
        if (baggage != null)
        {
            var header = baggage.ToHeader();
            if (!string.IsNullOrEmpty(header))
            {
                headers["baggage"] = header;
            }
        }
    }

    public static (TraceContext? Context, Baggage Baggage) ExtractContext(IDictionary<string, string> headers)
    {
        TraceContext? ctx = null;
        if (headers.TryGetValue("traceparent", out var traceparent))
        {
            ctx = TraceContext.FromTraceparent(traceparent);
        }

        var baggage = headers.TryGetValue("baggage", out var baggageHeader)
            ? Baggage.FromHeader(baggageHeader)
            : new Baggage();

        return (ctx, baggage);
    }
}
